pkg/manager/manager.go

```
@@ -670,8 +770,87 @@ func UpgradeRule(cursorDir string, ruleKey string) error {
+			if err != nil {
+				return fmt.Errorf("failed to upgrade GitHub reference: %w", err)
+			}
+
+			return nil
+		} else {
+			// For pinned commits, we would normally not update (it's pinned)
+			// But we'll re-download the file in case something went wrong
+			fmt.Printf("Rule %q is pinned to commit %s\n", ruleKey, shortCommit(gitRef))
+			fmt.Print("Do you want to re-download this pinned version? (y/N): ")
+			var answer string
+			fmt.Scanln(&answer)
```

golangci-lint (1.64.8)

Error return value of `fmt.Scanln` is not checked

(errcheck)